src/services/expensesService.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+let expenses = [];
+
+const clearExpenses = () => {
+  expenses = [];
+};
+
+const getAll = ({ userId, categories, from, to }) => {
+  if (!userId && !categories && !from && !to) {
+    return expenses;
+  }
+
+  return expenses.filter((expense) => {
+    if (userId && +userId !== expense.userId) {
+      return false;
+    }
+
+    if (categories && !categories.includes(expense.category)) {
```

Comment:
if there are no `categories` in search params we will get `undefined`
if only 1 `categories` we will get just value (....?categories=1)
And if we have many categories we will get array of values (....?categories=1&categories=2&categories=3)

![image](https://github.com/mate-academy/node_accounting-app-v2/assets/86789161/ea010564-92a2-4116-9fcf-a66739d6d01f)

You code will crash if you have only 1 category

Reply:
![image](https://github.com/mate-academy/node_accounting-app-v2/assets/43435306/0a990fe7-ed0d-4907-baa8-c162915a02a1)

Reply:
But what if the user makes call with `categories=123`?
And you have an item with the value `12` on DB
You will return this value to the user, but that's wrong